packages/design-system/src/components/Button/Button.stories.tsx

```diff
@@ -12,185 +13,187 @@ export default {
 	},
 };
 
-export const Primary = (props: Story) => {
-	return <Button.Primary {...props}>Basic Button</Button.Primary>;
+export const Primary = {
+	render: (props: Story<ButtonProps>) => {
+		return <Button.Primary {...props}>Basic Button</Button.Primary>;
+	},
 };
 
-export const PrimaryIcon = (props: Story) => {
+export const PrimaryIcon = (props: Story<Story<ButtonProps>>) => {
```

Comment:
```suggestion
export const PrimaryIcon = (props: Story<Story<ButtonProps>>) => {
```
What? If that's necessary then it's very bad. A generic loop is never a good sign.